src/Builder.php

```diff
@@ -262,14 +269,39 @@ public function useDismax()
     {
         $this->useDismax = true;
 
+        if (strpos($this->query, '*') !== false) {
+            $this->useExtendedDismax = true;
+        }
+
         return $this;
     }
 
+    /**
+     * Inform the builder that we want to use the *enhanced* dismax parser when building the query.
+     *
+     * @return $this
+     */
+    public function useEDismax()
+    {
+        $this->useDismax();
+        $this->useExtendedDismax = true;
+
+        return $this;
+    }
+
+    /**
+     * @return bool
+     */
+    public function isEDismax()
+    {
+        return $this->useExtendedDismax;
+    }
+
     /**
      * @return bool
      */
     public function isDismax()
     {
-        return $this->useDismax;
+        return $this->useDismax xor $this->useExtendedDismax;
```

Comment:
I'm not following your logic for using `xor` vs `||` here. Did you want this to return false if both `useDismax` and `useExtendedDismax` are true? If so, why?

Just for clarity - not challenging the choice here, just want to understand it.

Reply:
Actually this was a tough decision to make and I'm open new ideas.

>  Did you want this to return false if both useDismax and useExtendedDismax are true?

That's exactly what I'm trying to do.

Ultimately the `isEDismax` function is currently only used within the `SolrEngine.php` file to trigger `$query->getEDismax()`. 

```
$query = $this->client->createSelect();
        if ($builder->isDismax()) {
            $dismax = $query->getDisMax();
        } elseif ($builder->isEDismax()) {
            $dismax = $query->getEDisMax();
        }
```

I could have gone with `||` inside the `Builder.php` file and then checked for `EDismax` first and follow it with an `elseif` for `Dismax` within the `SolrEngine.php` file, but it felt as if it was unclear enough. 


That's what ultimately drove my decision to have only one of the `is(E)Dismax` functions return `true`.

Having said that, I must admit that I saw no other good solution for the time being. Please let me know if you have any suggestion as in how it should be changed.

Thank you!

Reply:
After a second look and further reflection, I changed the code to further split the `useDismax` from `useExtendedDismax` variables. Please let me know what you think now.

Reply:
I'm happy with it, thanks for talking it through with me. I'll get it merged and a new release out ASAP.

Reply:
Thank you as well! This has been a pleasure!